sdk/appcenter-storage/src/test/java/com/microsoft/appcenter/storage/StorageTest.java

```diff
@@ -484,6 +486,53 @@ public void readWithoutNetwork() {
                 any(ReadOptions.class));
     }
 
+    @Test
+    public void readWhenLocalStorageContainsDeletePendingOperation() {
+        when(mNetworkStateHelper.isNetworkConnected()).thenReturn(false);
+        when(SharedPreferencesManager.getString(PARTITION_NAME)).thenReturn(tokenResult);
+        Document<String> deletedDocument = new Document<>();
+        deletedDocument.setPendingOperation(Constants.PENDING_OPERATION_DELETE_VALUE);
+        when(mLocalDocumentStorage.read(anyString(), anyString(), any(Class.class), any(ReadOptions.class))).thenReturn(deletedDocument);
+        Document<String> document = Storage.read(PARTITION_NAME, DOCUMENT_ID, String.class).get();
+        assertNotNull(document.getDocumentError());
+        assertTrue(document.getDocumentError().getError().getMessage().contains("deleted"));
```

Comment:
I'm not a big fan of checking the contents of the message, but I can see why we need to test it.  Is it possible to additionally check the exception type (`StorageException`) or do we not have access to that?

Reply:
If it's possible, it would be good to add a similar check for the Storage Exception for a partition name that's not found.

Reply:
Let's check only for exception type here.

Reply:
changed! just for exception type